spec/classes/nscd_spec.rb

```diff
@@ -1,28 +1,86 @@
 require 'spec_helper'
 describe 'nscd' do
-  context 'with defaults for all parameters' do
-    it { is_expected.to contain_class('nscd::install') }
-    it { is_expected.to contain_class('nscd::config') }
-    it { is_expected.to contain_class('nscd::service') }
-    it { is_expected.to contain_package('nscd').with_ensure('present') }
-    it { is_expected.to contain_service('nscd').with_ensure('running') }
-  end
+  on_supported_os.each do |os, facts|
+    context "on #{os}" do
+      let(:facts) do
```

Comment:
you can also do: let :facts do

both versions work. You don't need to change it. I'm just a lazy person so I never use () here.